src/views/Search/useUpdateSearchParam.ts

```diff
@@ -0,0 +1,23 @@
+import { useCallback } from "react";
+import { useHistory } from "react-router-dom";
+import { getSearchPath } from "../../util/url";
+import { getSearchUrlParams } from "./util";
+
+export const useUpdateSearchParam = () => {
+  const { push } = useHistory();
+
+  return useCallback(
+    (p?: Partial<Parameters<typeof getSearchPath>[0]>) => {
```

Comment:
It's so that we can safely maintain the function reference returned by `useUpdateSearchParam` between renders, which will allow us to use it as a dependency in other hooks such as a `useEffect` or another `useCallback`

Reply:
Ohhh I see! 😮